kclvm/sema/src/ty/context.rs

```diff
@@ -52,6 +55,27 @@ impl TypeContext {
         is_cyclic_directed(&self.dep_graph)
     }
 
+    /// Add dependencies and return the dependency graph is cyclic.
+    pub fn add_dependencies(&mut self, from: &str, to: &str) -> bool {
+        let from_idx = self.get_node_index(from);
+        let to_idx = self.get_node_index(to);
+        self.dep_graph.add_edge(from_idx, to_idx, ());
+        self.is_cyclic()
+    }
+
+    /// Get the node index from the node index map or insert it into the dependency graph.
+    #[inline]
+    fn get_node_index(&mut self, name: &str) -> NodeIndex {
```

Comment:
get call will change the self, get or `midify` ...?

Reply:
> get call will change the self, use get or `get_or_insert_...` name?

I will fix it on another PR.

Reply:
> get call will change the self, use get or `get_or_insert_...` name?

fixed.